discovery_engine/test/integration/active_search_test.dart

```diff
@@ -0,0 +1,133 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:io' show Directory;
+
+import 'package:test/test.dart';
+
+import 'package:xayn_discovery_engine/discovery_engine.dart'
+    show
+        ActiveSearchRequestFailed,
+        ActiveSearchRequestSucceeded,
+        ClientEventSucceeded,
+        DiscoveryEngine,
+        RestoreActiveSearchFailed,
+        SearchFailureReason;
+import 'package:xayn_discovery_engine/src/api/api.dart';
+
+import '../logging.dart' show setupLogging;
+import 'utils/helpers.dart'
+    show TestEngineData, initEngine, setupTestEngineData;
+import 'utils/local_newsapi_server.dart' show LocalNewsApiServer;
+
+void main() {
+  setupLogging();
+
+  group('DiscoveryEngine active search test', () {
+    late LocalNewsApiServer server;
+    late TestEngineData data;
+    late DiscoveryEngine engine;
+
+    setUp(() async {
+      data = await setupTestEngineData();
+      server = await LocalNewsApiServer.start();
+      engine = await initEngine(data, server.port);
+      expect(engine, isA<DiscoveryEngine>());
+    });
+
+    tearDown(() async {
+      await engine.dispose();
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('basic search works', () async {
+      final response = await engine.requestQuerySearch('some search query');
+      expect(response, isA<ActiveSearchRequestSucceeded>());
+      expect(
+        (response as ActiveSearchRequestSucceeded).items,
+        isNotEmpty,
+      );
+    });
+
+    test('restoring when no active search is running returns an error',
+        () async {
+      final response = await engine.restoreActiveSearch();
+      expect(response, isA<RestoreActiveSearchFailed>());
+      expect(
+        (response as RestoreActiveSearchFailed).reason,
+        equals(SearchFailureReason.noActiveSearch),
+      );
+    });
+
+    test('restoring a search works', () async {
+      // In order to restore a search, we first need to initiate one
+      final searchResponse =
+          await engine.requestQuerySearch('some search query');
+      expect(searchResponse, isA<ActiveSearchRequestSucceeded>());
+      final items = (searchResponse as ActiveSearchRequestSucceeded).items;
+      expect(items, isNotEmpty);
+
+      // Let's restore the search and see if it's the same we just sent
+      final restoreResponse = await engine.restoreActiveSearch();
+      expect(restoreResponse, isA<RestoreActiveSearchSucceeded>());
+      final succeededResponse = restoreResponse as RestoreActiveSearchSucceeded;
+
+      expect(succeededResponse.search.searchTerm, equals('some search query'));
+      expect(succeededResponse.items, equals(items));
+    });
+
```

Comment:
🔧  In TY-2698 there is also the task to test that `closeActiveSearch` returns an error when no active search exists

Reply:
Good point. I removed that test again, but forgot to record that. I removed it because it does not currently fail. I added it back, and it's currently skipped.